### PR TITLE
chore(common): allow build agents to automatically select emsdk version, and enable support for 3.1.60+

### PR DESCRIPTION
**Pull request by mcdurdin**
The build agents will now automatically install and activate the Emscripten version found in minimum-versions.inc.sh when configuring WASM projects. For developer machines, this behaviour can be enabled by setting the environment variable `KEYMAN_USE_EMSDK` (recommended).

This will be back-ported to 17.0 so that we can ensure that all builds use a consistent Emscripten version.

Also, Emscripten 3.1.60 made a breaking change to `BindingType::toWireType` (why is this done in a patch version?) so this change ensures that we can continue to build on 3.1.58 and 3.1.64 (which is needed for #12234), as a bridging strategy.

Relates-to: emscripten-core/emscripten#21692
Relates-to: #12234

@keymanapp-test-bot skip

### CODE REVIEW

**Review by darcywong00 — APPROVED** [review]
lgtm

File: resources/build/minimum-versions.inc.sh [hunk 1]
```diff
@@ -20,8 +20,7 @@ KEYMAN_MIN_TARGET_VERSION_CHROME=95.0         # Final version that runs on Andro
 # Dependency versions
 KEYMAN_MIN_VERSION_NODE_MAJOR=20              # node version source of truth is /package.json:/engines/node
 KEYMAN_MIN_VERSION_NPM=10.5.1                 # 10.5.0 has bug, discussed in #10350
-KEYMAN_MIN_VERSION_EMSCRIPTEN=3.1.44          # Warning: 3.1.45 is bad (#9529); newer versions work
-KEYMAN_MAX_VERSION_EMSCRIPTEN=3.1.58          # See #9529
+KEYMAN_MIN_VERSION_EMSCRIPTEN=3.1.58          # Use KEYMAN_USE_EMSDK to automatically update to this version
```
Comment (darcywong00) [inline resources/build/minimum-versions.inc.sh:23]: Does running `publish-minimum-versions.sh` still work? (The Note on l. 8)

Comment (mcdurdin) [reply resources/build/minimum-versions.inc.sh:23]: Yes, oops will run in a follow-up PR. Getting this merged so we can unbreak builds.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.95-alpha](https://keyman.com/downloads/releases/alpha/18.0.95)

**Pull request merged**